src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerFindMatches.java

```diff
@@ -83,39 +86,46 @@ public AboutToStartOrSubmitResponse<CaseData, State> aboutToStart(final CaseDeta
 
     }
 
-    // Helper method to create term queries with variations for spaces
-    private BoolQueryBuilder createNameMatchQuery(String field, String name) {
-        return QueryBuilders.boolQuery()
-            .should(QueryBuilders.termQuery(field, name)) // Exact match
-            .should(QueryBuilders.termQuery(field, " " + name)) // Prepend space
-            .should(QueryBuilders.termQuery(field, name + " ")) // Append space
-            .should(QueryBuilders.termQuery(field, " " + name + " ")); // Prepend and append space
+    public String[] normalizeAndSplit(String name) {
+        // remove "name changed by Deed Poll" ignoring case
+        String nameWithoutDeedPollStatement = name.replaceAll("(?i)name changed by deed poll", "").trim();
```

Comment:
It's a bit of a nitpick, but I suppose we could have variations of this, e.g. "changed by deed poll". Another way we could do it is to look for any text in brackets that includes "deed poll" although it might not be in brackets and there's a risk we could strip out a real name if a bracket was included at the start of the input by mistake. Handling all the scenarios gets very complex and I'm not sure what's best

Reply:
if they put in exactly as on uk marriage cert then this should be ok, if they don't then it will match against each word which likely won't hit a lot of names, maybe the 'by' and 'poll' with letby or pollard or somesuch.